matchers/compat/parse.go

```diff
@@ -0,0 +1,218 @@
+// Copyright 2023 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package compat
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/go-kit/log"
+	"github.com/go-kit/log/level"
+
+	"github.com/prometheus/alertmanager/featurecontrol"
+	"github.com/prometheus/alertmanager/matchers/parse"
+	"github.com/prometheus/alertmanager/pkg/labels"
+)
+
+var (
+	parseMatcher  = stableMatcherParser(log.NewNopLogger())
+	parseMatchers = stableMatchersParser(log.NewNopLogger())
+)
+
+type matcherParser func(s string) (*labels.Matcher, error)
+
+type matchersParser func(s string) (labels.Matchers, error)
+
+// Matcher parses the matcher in the input string. It returns an error
+// if the input is invalid or contains two or more matchers.
+func Matcher(s string) (*labels.Matcher, error) {
+	return parseMatcher(s)
+}
+
+// Matchers parses one or more matchers in the input string. It returns
+// an error if the input is invalid.
+func Matchers(s string) (labels.Matchers, error) {
```

Comment:
I went on a hunt to try and understand why we weren't using this function. On that mission, I concluded that we should address the inconsistency I see between the direct use of the functions in the `compat` package  and `utils.go` functions

Reply:
If we're still not using this function after that refactor - I believe we should remove it.

Reply:
This will be used in Alertmanager server, both in `api/v1/api.go` and `config/config.go`. I'm surprised it's not used in `api/v2/` though, I need to look into that.